src/main/java/com/poiji/bind/mapping/XSSFUnmarshaller.java

```diff
@@ -39,6 +42,12 @@
 
     <T> void unmarshal0(Class<T> type, Consumer<? super T> consumer, OPCPackage open) throws IOException, SAXException, OpenXML4JException {
 
+        //ISSUE #55
+        XSSFWorkbook wb = new XSSFWorkbook(open);
```

Comment:
It kinda makes sense as to why it does that, but that's annoying.  At the moment I am using my version and we are not importing that number of rows, so I would like to try and find a way to fix this before forced to limit only to xls.  So unless there are other developers that need this functionally now, can you hold of on doing the pull and I will do some investigating and let you know what I find.

Reply:
no problem, I will inform you as well if I find something useful for this case.

Reply:
Hey, I have done some work on a different way to find hidden and visible sheets to avoid the GC exception.  Can you please have a look at it when you get a chance and see if its an ok way to do it, and if it makes sense and if passes your 187k test.  Have also included some big excel files for tests, hope they are not to large for repo.

The code still needs to be tidied up and made neater but just would like you to have a look first.  If you happy with it I will go and tidy it all up and then can merge.

Thanks

Reply:
Hi @Developer-Matt , good news is that all tests pass successfully yay! :) , I will look at the changes you have made and will make some changes If necessary.